### ISSUE TITLE
Problem of reproducing paper results

### ISSUE THREAD

**Issue by zzzmm1** [GISec-Team/NCGNN#1]
Thanks for your outstanding work!
I refer to 4.2 Code Execution in readme:
python main.py -D computers -M NCGCN
But the test_acc is only 57, which is far from the results of the paper. 
May I ask if there is any hyperparameter setting that is incorrect?

**Comment by PaintingProgrammer** [comment]
Hi, thanks for your feedback!

The difference in test accuracy could be due to the following factors:

1. **Hyperparameter Tuning:** I recommend tuning the hyperparameters based on the space mentioned in section 4.3 of the README. You can also consider using tools like NNI for automated hyperparameter optimization. Additionally, we mention in the paper that the hidden layer dimension for the Computers dataset is 512, while the default value in the code is set to 64.
2. **Hardware/Environment:** The optimal parameters are provided in the paper's appendix, but performance may vary due to differences in hardware (e.g., GPU) and software versions (e.g., CUDA).
3. **Experimental Setup:** Different random seeds can also affect the results. In the paper, we used seeds from [42, 43, 44, 45, 46, 47, 48, 49, 50, 51] for 10 repetitions, so please make sure you're using the same random seeds to match the paper's results. Additionally, we used a 60%/20%/20% split for training/validation/testing, so ensure your data split matches that as well.

Hope this helps!

**Comment by zzzmm1** [comment]
Hi, thanks for your reply.
I have changed the hidden layer dimension to 512, and keep the other hyperparameter as provided in the paper's appendix.
There is my training Namespace:
Namespace(act='relu', addself=0, baseseed=42, dataset='computers', dp1=0.2, dp2=0.6, forcing=0, hidden=512, hops=2, lr=0.05, model='NCGCN', threshold=1.1748975549395295, wd=0.05)
There is a question,  is the threshold in the appendix 0.7, which is the value calculated after line 134 [main.py](https://github.com/GISec-Team/NCGNN/blob/main/main.py) or is it an input value?
The seeds and split ratio are also same as yours.
Could you provide the hyperparameters optimized on the Computers dataset to facilitate the reproduction of the results in the paper？

**Comment by LinusLing** [comment]
Any updates? same issue.

**Comment by rockcor** [comment]
> Hi, thanks for your reply. I have changed the hidden layer dimension to 512, and keep the other hyperparameter as provided in the paper's appendix. There is my training Namespace: Namespace(act='relu', addself=0, baseseed=42, dataset='computers', dp1=0.2, dp2=0.6, forcing=0, hidden=512, hops=2, lr=0.05, model='NCGCN', threshold=1.1748975549395295, wd=0.05) There is a question, is the threshold in the appendix 0.7, which is the value calculated after line 134 [main.py](https://github.com/GISec-Team/NCGNN/blob/main/main.py) or is it an input value? The seeds and split ratio are also same as yours. Could you provide the hyperparameters optimized on the Computers dataset to facilitate the reproduction of the results in the paper？

The threshold in the printed Namespace should from 0-1 with 0.1 interval. @zzzmm1

**Comment by rockcor** [comment]
<img width="2031" height="1162" alt="Image" src="https://github.com/user-attachments/assets/527d89d8-d590-4a47-926c-fbd6eb0c74f9" />
Also the wd should be 0.00005.
@zzzmm1 @LinusLing
